### PR TITLE
Reactivate some XO rules

### PR DESCRIPTION
**Pull request by sindresorhus**

### CODE REVIEW

File: renderer/containers/preferences.js [hunk 1]
```diff
@@ -123,6 +123,8 @@ export default class PreferencesContainer extends Container {
   }
 
   toggleSetting = (setting, value) => {
+    // TODO: Fix this ESLint violation
+    // eslint-disable-next-line react/no-access-state-in-setstate
```
Comment (karaggeorge) [inline renderer/containers/preferences.js]: I still don't get why this is bad if we are talking about a "toggle" function. I guess a solution would be to pass in the `!state.value` from where it's called

Comment (sindresorhus) [reply renderer/containers/preferences.js]: @karaggeorge React state can be async, so the state might be changed between `this.state[setting]` is used and `this.setState({[setting]: newVal})` is finished.

Comment (albinekb) [reply renderer/containers/preferences.js]: Can you use `this.setState(state => nextState)`?

Comment (sindresorhus) [reply renderer/containers/preferences.js]: Yes, using the callback version of setState is the correct way to fix this. It ensures the updates are atomic.

Comment (karaggeorge) [reply renderer/containers/preferences.js]: Does unstated support that?

Comment (albinekb) [reply renderer/containers/preferences.js]: Yes @karaggeorge https://github.com/jamiebuilds/unstated#setstate

```setState() in Container mimics React's setState() method as closely as possible.
```

**Pull request merged**